modules/HumanStateVisualizer/src/main.cpp

```diff
@@ -0,0 +1,259 @@
+    }
+
+    // initialize state variables for visualization
+    iDynTree::Transform wHb = iDynTree::Transform::Identity();
+    iDynTree::VectorDynSize joints(viz.modelViz("human").model().getNrOfDOFs());
+    joints.zero();
+
+    // start Visualization
+    std::chrono::steady_clock::time_point now = std::chrono::steady_clock::now();
+    std::chrono::steady_clock::time_point lastViz = std::chrono::steady_clock::now();
+
+    long minimumMicroSecViz = std::round(1e6 / (double) maxVisualizationFPS);
+
+    while(viz.run())
+    {
+        now = std::chrono::steady_clock::now();
+        if (std::chrono::duration_cast<std::chrono::microseconds>(now - lastViz).count() < minimumMicroSecViz)
+        {
+            std::this_thread::sleep_for(std::chrono::milliseconds(1));
+            continue;
+        }
+
+        // read values from iHumanState
+        std::array<double, 3> basePositionInterface = iHumanState->getBasePosition();
+        std::array<double, 4> baseOrientationInterface = iHumanState->getBaseOrientation();
+        std::vector<double> jointPositionsInterface = iHumanState->getJointPositions();
+        std::vector<std::string> jointNames = iHumanState->getJointNames();
+        std::string baseName = iHumanState->getBaseName();
+
+        iDynTree::Vector4 baseOrientationQuaternion;
+        baseOrientationQuaternion.setVal(0, baseOrientationInterface.at(0));
+        baseOrientationQuaternion.setVal(1, baseOrientationInterface.at(1));
+        baseOrientationQuaternion.setVal(2, baseOrientationInterface.at(2));
+        baseOrientationQuaternion.setVal(3, baseOrientationInterface.at(3));
+
+        iDynTree::Position basePosition;
+        basePosition.setVal(0, basePositionInterface.at(0));
+        basePosition.setVal(1, basePositionInterface.at(1));
+        basePosition.setVal(2, basePositionInterface.at(2));
+
+        wHb.setRotation(iDynTree::Rotation::RotationFromQuaternion(baseOrientationQuaternion));
+        wHb.setPosition(basePosition);
+
+        for (size_t iHumanInterfaceIdx = 0; iHumanInterfaceIdx < jointNames.size(); iHumanInterfaceIdx++)
+        {
+            std::string jointName = jointNames.at(iHumanInterfaceIdx);
+            double jointVal = jointPositionsInterface.at(iHumanInterfaceIdx);
+
+            iDynTree::JointIndex jointIndex = viz.modelViz("human").model().getJointIndex(jointName);
+            if (jointIndex != iDynTree::JOINT_INVALID_INDEX)
+            {
+                joints.setVal(jointIndex, jointVal);
+            }
+        }
+
+        // follow the desired link with the camera
+        if ( !useFixedCamera )
+        {
+            viz.camera().setPosition(basePosition + cameraDeltaPosition);
+            viz.camera().setTarget(basePosition);
+        }
+        
+        
+
+        // Update the visulizer
+        viz.modelViz("human").setPositions(wHb, joints);
+        viz.draw();
+        lastViz = std::chrono::steady_clock::now();
+    }
+
+    remapperDevice.close();

```

Comment:
Just FYI, if you launch this application from the yarpmanager, when you stop it (again from the yarpmanager), it does not reach this part for some reason. On the other hand, if you close it from the "x" button on the Window, it closes cleanly.

Reply:
> when you stop it (again from the yarpmanager), it does not reach this part for some reason.

Yes, same is happening when ctrl-c. I think that is expected since for the time being it is a simple C++ application, and no interrupt catch is implemented.

Using the yarp `RFModule` and properly implementing the `close()` method would solve it, but for the time-being closing the windows should be the proper way of closing.

Reply:
> Using the yarp RFModule and properly implementing the close() method would solve it, but for the time-being closing the windows should be the proper way of closing.

Yes, even if the ``updateModule`` method is on another thread, so it may not be viable for the visualization. Maybe il would be better to catch the corresponding signal.

Reply:
For an example of a simple multiplatform Ctrl+C catching application, see https://github.com/traversaro/blockrunner/blob/master/src/BlockRunner/blockrunnerserver.cpp (feel free to ignore the context : ) ). Interestingly, I could not find some easy to use a simple C++ library to do that, something like https://github.com/Detegr/rust-ctrlc but for C++, that may be another niche to fill.

Reply:
Thanks @traversaro, with https://github.com/robotology/human-dynamics-estimation/pull/230/commits/235ad59bfa9160f1537338f93ee9c9b9da7e67bc I replicate the code you pointed and it results working fine.